### ISSUE TITLE
UnsuccessfulResponse exception is not thrown when using preventUnsuccessfulResponse method

### ISSUE THREAD

**Issue by phuclh** [stefanzweifel/sidecar-browsershot#25]
In the Spatie browsershot package, when using `preventUnsuccessfulResponse` method, it will throw `UnsuccessfulResponse` exception:

```
$browsershot->preventUnsuccessfulResponse($this->preventUnsuccessfulResponse);
```

But in this package, it throws `Hammerstone\Sidecar\Exceptions\LambdaExecutionException`, so it is hard to detect if a URL has HTTP error respond code like `400, 500, etc`.

Thank you!

**Comment by stefanzweifel** [comment]
I have to take a closer look at this. 
The general problem with `sidecar-browsershot` seems to be, that users expect that Spatie's exceptions are thrown.

As mentioned in #19, maybe you can manually inspect the LambdaExectionException and in your app and act accordingly?

**Comment by phuclh** [comment]
@stefanzweifel Yeah, that is what I am using now. I just took a quick look. Seems in the BrowsershotLambda, you threw all exception with this method:

<img width="434" alt="CleanShot 2022-08-11 at 16 24 57@2x" src="https://user-images.githubusercontent.com/6707194/184258698-ca2e3890-e217-42fa-a6cb-71a25a8fee12.png">

While in the Spatie package, it checks by the exit code:
<img width="911" alt="CleanShot 2022-08-11 at 16 26 11@2x" src="https://user-images.githubusercontent.com/6707194/184258808-37c93b86-2271-4f90-9fea-4b308e81265d.png">

**Comment by stefanzweifel** [comment]
This seems to be more complicated than I thought.

Ideally, we could reuse the exit codes from Spatie's [`browser.js`](https://github.com/spatie/browsershot/blob/9e93f0cb9e76e85a9ed61723d2d93beab57345aa/bin/browser.js#L366-L389)-code and do the same `$exitCode` check in `BrowsershotLambda@callBrowser` as Spatie does.

Right now, we do not differentiate between errors and body HTML [here](https://github.com/stefanzweifel/sidecar-browsershot/blob/3a93c9fa25afa4cbe3800993b44d3401d53285dc/resources/lambda/browsershot.js#L94). 
(Maybe I'm wrong and Sidecar already catches possible JavaScript/Node errors thrown in `execSync`. Then, `result.toString()` would always be the body HTML of a page. @aarondfrancis Do you have any insights about this?)

@phuclh You mention that you manually inspect the `LambdaExecutionException`. What exactly are you doing there right now? Do you re-throw the Spatie Exceptions? Are you able to share some code?
